packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -459,7 +459,7 @@ class IrisGridTableModelTemplate<
 
   get isDeletable(): boolean {
     if (this.inputTable !== null) {
-      return this.inputTable?.keyColumns.length > 0;
+      return this.keyColumnSet.size > 0;
     }
     return false;
```

Comment:
Don't need to check if `this.inputTable != null`, we can just do `return this.keyColumnSet.size > 0`